monitor/inclusion_checker.go

```diff
@@ -0,0 +1,156 @@
+package monitor
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/tls"
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/ct-woodpecker/storage"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var oldestUnseen = promauto.NewGauge(prometheus.GaugeOpts{
+	Name: "oldest_unincorporated_cert",
+	Help: "Number of seconds since the oldest unincorporated certificate was submitted",
+})
+
+type InclusionOptions struct {
+	Interval       time.Duration
+	FetchBatchSize int64
+}
+
+type inclusionChecker struct {
+	logger           *log.Logger
+	client           monitorCTClient
+	logURI           string
+	db               storage.Storage
+	signatureChecker *ct.SignatureVerifier
+	clk              clock.Clock
+	logID            int64
+
+	interval  time.Duration
+	batchSize int64
+}
+
+func (ic *inclusionChecker) run() {
+	go func() {
+		for {
+			err := ic.checkInclusion()
+			if err != nil {
+				ic.logger.Printf("!!! Checking certificate inclusion failed: %s", err)
+			}
+			ic.clk.Sleep(ic.interval)
+		}
+	}()
+}
+
+func (ic *inclusionChecker) checkInclusion() error {
+	current, err := ic.db.GetIndex(ic.logID)
+	if err != nil {
+		return fmt.Errorf("error getting current log index: %s", err)
```

Comment:
can you include the `ic.logID` in this error message (and maybe in the other `fmt.Errorf`'s)?